submission/proxy_server.go

```diff
@@ -45,6 +46,20 @@ func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout
 // Run starts regular Log list updates. Blocks until initialization happens.
 func (s *ProxyServer) Run(logListRefreshInterval time.Duration, rootsRefreshInterval time.Duration) {
 	s.p.Run(context.Background(), logListRefreshInterval, rootsRefreshInterval)
+
+	// start listening to proxy errs
+	go func() {
+		for {
+			select {
+			case <-context.Background().Done():
```

Comment:
`context.Background()` should only be called at the top level, and always passed into functions, that way it can be canceled etc.  Calling `context.Background()` here means that it will never be canceled and is a bit of a no-op.

Reply:
Added context argument.